modules/candidate_profile/php/module.class.inc

```diff
@@ -0,0 +1,88 @@
+<?php
+namespace LORIS\candidate_profile;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\StudyEntities\Candidate\CandID;
+
+/**
+ * Class module implements the basic LORIS module functionality
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class Module extends \Module
+{
+    /**
+     * The candidate_profile module overrides the default routing to ensure that
+     * /candidate_profile/123456 goes to the candidate_profile page for candidate
+     * 123456.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $candIDStr = ltrim($request->getURI()->getPath(), '/');
+        try {
+            $candID    = new CandID($candIDStr);
+            $candidate = \Candidate::singleton($candID);
+
+            $request = $request->withAttribute('candidate', $candidate);
+            $page    = $this->loadPage("candidate_profile");
+            return $page->process($request, $page);
+
+        } catch (\DomainException | \LORISException $e) {
+            // A LORISException means the \Candidate::singleton couldn't
+            // load the candidate (ie because the CandID doesn't exist
+            // in the database.) A \DomainException means the CandID format
```

Comment:
If the CandID format is invalid shouldn't that be a 400 instead of a 404?

Reply:
No, if the CandID format is not valid that means they accessed a url like '/candidate_profile/abc123'. The URL doesn't exist, it's not found.